src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessorLinux.cs

```diff
@@ -0,0 +1,199 @@
+                if (filePath != null && File.Exists(filePath))
+                {
+                    Logger.LogError("Cannot create MemoryMappedFile: {MapName}, file already exists", mapName);
+                    return false;
+                }
+
+                // Get path of where to create the new file-based MemoryMappedFile
+                filePath = CreatePath(mapName, size);
+                if (string.IsNullOrEmpty(filePath))
+                {
+                    Logger.LogError("Cannot create MemoryMappedFile: {MapName}, invalid file path", mapName);
+                    return false;
+                }
+
+                mmf = MemoryMappedFile.CreateFromFile(
+                    filePath,
+                    FileMode.OpenOrCreate,
+                    null, // Named maps are not supported on Linux
+                    size,
+                    MemoryMappedFileAccess.ReadWrite);
+                return true;
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Cannot create MemoryMappedFile: {mapName} with size: {Size} bytes", mapName, size);
+            }
+
+            return false;
+        }
+
+        public override bool TryOpen(string mapName, out MemoryMappedFile mmf)
+        {
+            mmf = null;
+
+            try
+            {
+                string filePath = GetPath(mapName);
+                if (filePath != null && File.Exists(filePath))
+                {
+                    mmf = MemoryMappedFile.CreateFromFile(filePath, FileMode.Open);
+                    return true;
+                }
+                else
+                {
+                    Logger.LogDebug("Cannot open file path: {FilePath}, path not found", filePath);
+                    return false;
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Cannot open MemoryMappedFile: {MapName}", mapName);
+                return false;
+            }
+        }
+
+        public override void Delete(string mapName, MemoryMappedFile mmf)
+        {
+            if (mmf != null)
+            {
+                mmf.Dispose();
+            }
+
+            try
+            {
+                string filePath = GetPath(mapName);
+                File.Delete(filePath);
+            }
+            catch (Exception e)
+            {
+                Logger.LogWarning(e, "Cannot delete MemoryMappedFile: {MapName}", mapName);
+            }
+        }
+
+        /// <summary>
+        /// Get the path of the file to store a <see cref="MemoryMappedFile"/>.
+        /// We first try to mount it in memory-mounted directories (e.g. /dev/shm/).
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/>.</param>
+        /// <returns>Path to the <see cref="MemoryMappedFile"/> if found an existing one
+        /// <see cref="null"/> otherwise.</returns>
+        public string GetPath(string mapName)
```

Comment:
Review accessibility of GetPath/CreatePath/GetDirectory - shouldn't they be private helpers?